### PR TITLE
[CHORE] connect: Optimize plans in connect

### PR DESCRIPTION
**Pull request by colin-ho**
1. Optimize logical plans before executing them in connect.
2. Run the native executor on a spawned task on the connect runtime instead of a dedicated thread. The native executor already spawns it's own thread, and it also returns an awaitable receiver, so we should be able to just receive from the connect runtime.

### CODE REVIEW

File: src/daft-local-execution/src/channel.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@ impl<T> Receiver<T> {
     pub(crate) fn blocking_recv(&self) -> Option<T> {
         self.0.recv().ok()
     }
+
+    pub(crate) fn into_inner(self) -> loole::Receiver<T> {
```
Comment (andrewgazelka) [inline src/daft-local-execution/src/channel.rs:20]: question: does loole have great perf characterisics? why are we using it?

File: src/daft-connect/src/op/execute/root.rs [hunk 2]
```diff
@@ -28,36 +28,32 @@ impl Session {
 
         let finished = context.finished();
 
-        let (tx, rx) = tokio::sync::mpsc::channel::<eyre::Result<ExecutePlanResponse>>(16);
-        std::thread::spawn(move || {
-            let result = (|| -> eyre::Result<()> {
+        let (tx, rx) = tokio::sync::mpsc::channel::<eyre::Result<ExecutePlanResponse>>(1);
+        tokio::spawn(async move {
```
Comment (andrewgazelka) [inline src/daft-connect/src/op/execute/root.rs:32]: in future this could just be a stream but I think this is good for now.

**Review by andrewgazelka — APPROVED** [review]
lgtm

File: src/daft-local-execution/src/channel.rs (hunk 1 above)
Comment (colin-ho) [reply src/daft-local-execution/src/channel.rs:20]: Performance wise it's decently better, especially for single capacity channels (which is what most of the executor channels are). It's also a MPMC channel, which tokio doesn't provide.

File: src/daft-connect/src/op/execute/root.rs (hunk 2 above)
Comment (colin-ho) [reply src/daft-connect/src/op/execute/root.rs:32]: Yup, agreed

**Pull request merged**